Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.GotoExpression.cs

```diff
@@ -68,8 +68,11 @@ protected internal override Expression Accept (ExpressionVisitor visitor)
       return default(Expression);
     }
 
+    [Pure]
     public GotoExpression Update (LabelTarget target, Expression value)
     {
+      Contract.Requires(target != null);
+      Contract.Ensures(Contract.Result<GotoExpression>() != null);
```

Comment:
Following requires could be added:

`Contract.Requies(value != null || target.Type == typeof(void), "Label must be void or have an Expression");`

From http://referencesource.microsoft.com/#System.Core/Microsoft/Scripting/Ast/GotoExpression.cs,357